virttest/qemu_devices/qdevices.py

```diff
@@ -842,6 +842,10 @@ def get_qid(self):
     def _cmdline_json(self):
         params = self.params.copy()
         out = "-%s " % self.type
+        for key, val in params.items():
+            # "cache-size" is from device( "driver": "qcow2" )
+            if key == "cache-size":
+                params[key] = int(val)
```

Comment:
Suggest letting the subclass `QBlockdevFormatQcow2` override this method `_cmdline_json` since this change relates to the `QBlockdevFormatQcow2`.
And I think there are also other blockdev nodes has such a change in the future, so we just focus on updating the corresponding subclass of the method instead of updating the parent class method.

Reply:
> Suggest letting the subclass `QBlockdevFormatQcow2` override this method `_cmdline_json` since this change relates to the `QBlockdevFormatQcow2`. And I think there are also other blockdev nodes has such a change in the future, so we just focus on updating the corresponding subclass of the method instead of updating the parent class method.

Hello @YongxueHong , thanks for comment. Once we do the everything at subclass, there'll be a redundant codes. In addition, as I known, if "cache-size" is integer in qcow2 format, it must be integer in other format.  
What do you think?

Reply:
> > Suggest letting the subclass `QBlockdevFormatQcow2` override this method `_cmdline_json` since this change relates to the `QBlockdevFormatQcow2`. And I think there are also other blockdev nodes has such a change in the future, so we just focus on updating the corresponding subclass of the method instead of updating the parent class method.
> 
> Hello @YongxueHong , thanks for comment. Once we do the everything at subclass, there'll be a redundant codes. In addition, as I known, if "cache-size" is integer in qcow2 format, it must be integer in other format. What do you think?

Hi @nickzhq 
I don't think it is a redundant since we just make up for the base class method.
I mean that we can change the param dict then call the bass class method, like `super(QBlockdevFormatQcow2, self)._cmdline_json()` in the `QBlockdevFormatQcow2`
BTW, we can refer to other opinions @zhencliu @luckyh

Reply:
> hi @nickzhq , I am afraid cache-size is a qcow2-specific attribute, what @YongxueHong means here is for a parent class QBlockdevNode, if we add cache-size there, it means all its children should have this attribute, but that is not the case, in your case, only qcow2 format node has the attribute, so we may better add it there

Thanks for explanation. The code is on changing.